lib/global-styles.php

```diff
@@ -535,40 +535,49 @@ function gutenberg_experimental_global_styles_normalize_schema( $tree ) {
 	return $tree;
 }
 
-/**
- * Returns the stylesheet resulting of merging
- * core's, theme's, and user's origins.
- *
- * @return string
- */
-function gutenberg_experimental_global_styles_get_stylesheet() {
-	$gs_merged = array();
-	$gs_core   = gutenberg_experimental_global_styles_get_core();
-	$gs_theme  = gutenberg_experimental_global_styles_get_theme();
-	$gs_user   = gutenberg_experimental_global_styles_get_user();
+function gutenberg_experimental_global_styles_get_merged_origins() {
+	$core   = gutenberg_experimental_global_styles_get_core();
+	$theme  = gutenberg_experimental_global_styles_get_theme();
+	$user   = gutenberg_experimental_global_styles_get_user();
+	$merged = gutenberg_experimental_global_styles_merge_trees( $core, $theme, $user );
 
-	$gs_merged = gutenberg_experimental_global_styles_merge_trees( $gs_core, $gs_theme, $gs_user );
-
-	$stylesheet = gutenberg_experimental_global_styles_resolver( $gs_merged );
-	if ( empty( $stylesheet ) ) {
-		return;
-	}
-	return $stylesheet;
+	return $merged;
 }
 
 /**
  * Fetches the preferences for each origin (core, theme, user)
  * and enqueues the resulting stylesheet.
  */
 function gutenberg_experimental_global_styles_enqueue_assets() {
-
-	$stylesheet = gutenberg_experimental_global_styles_get_stylesheet();
+	$merged     = gutenberg_experimental_global_styles_get_merged_origins();
+	$stylesheet = gutenberg_experimental_global_styles_resolver( $merged );
+	if ( empty( $stylesheet ) ) {
+		return;
+	}
 
 	wp_register_style( 'global-styles', false, array(), true, true );
 	wp_add_inline_style( 'global-styles', $stylesheet );
 	wp_enqueue_style( 'global-styles' );
 }
 
+/**
+ * Returns the default config for editor features,
+ * or an empty array if none found.
+ *
+ * @param array $config Config to extract values from.
+ * @return array Default features config for the editor.
+ */
+function gutenberg_experimental_global_styles_get_editor_features( $config ) {
+	if (
+		empty( $config['global']['features'] ) ||
+		! is_array( $config['global']['features'] )
+	) {
+		return array();
+	}
+
+	return $config['global']['features'];
```

Comment:
I don't really think we should pass just the "global" features to the editor, we should pass all the features config (per block as well) and handle it in useFeature properly.

Reply:
Yup, that's my next step.